File: database/booltostring.js
const AWS = require('aws-sdk');

AWS.config.update({ region: 'us-east-1' });


const dynamoDB = new AWS.DynamoDB.DocumentClient();

async function batchUpdateOngoingAttribute(items, batchSize) {
  for (let i = 0; i < items.length; i += batchSize) {
    const batch = items.slice(i, i + batchSize);
    const promises = batch.map(async (item) => {
      const updateParams = {
        TableName: 'event-data',
        Key: {
          'id': item['id'],
          // Include other primary key attributes if necessary
        },
        UpdateExpression: 'SET ongoing = :ongoingVal',
        ExpressionAttributeValues: {
          ':ongoingVal': item.ongoing ? 'true' : 'false',
        }
      };
      return dynamoDB.update(updateParams).promise();
    });

    try {
      await Promise.all(promises);
    } catch (error) {
      console.error('Error updating batch:', error);
      // Handle error, possibly retry failed updates
    }
  }
}

async function updateAllOngoingAttributes() {
  const params = {
    TableName: 'event-data',
    // Add any necessary parameters for scanning or querying your items
  };

  try {
    const data = await dynamoDB.scan(params).promise();
    await batchUpdateOngoingAttribute(data.Items, 25); // Update in batches of 10
  } catch (error) {
    console.error('Error scanning table:', error);
    throw error;
  }
}

// Usage
updateAllOngoingAttributes().then(() => {
  console.log('Successfully updated items.');
}).catch(error => {
  console.error('Error:', error);
});

File: database/booltostring.py
import boto3

# Initialize the DynamoDB client
dynamodb = boto3.client('dynamodb', region_name='us-east-1')

# Specify your DynamoDB table name
table_name = 'event-data'

# Define a function to update the boolean values in the table
def update_ongoing_values():
    # Use a scan operation to retrieve all items in the table
    response = dynamodb.scan(TableName=table_name)
    
    # Iterate through the items and update the 'ongoing' attribute
    for item in response['Items']:
        # Check if 'ongoing' attribute exists in the item
        if 'ongoing' in item:
            # Check if 'ongoing' attribute is a Boolean
            if 'BOOL' in item['ongoing']:
                # Get the current Boolean value
                ongoing_boolean = item['ongoing']['BOOL']
                
                # Convert the Boolean value to its string representation
                ongoing_string = str(ongoing_boolean).lower()
                
                # Update the item in the table with the string value
                dynamodb.update_item(
                    TableName=table_name,
                    Key={'id': item['id']},
                    UpdateExpression='SET ongoing = :value',
                    ExpressionAttributeValues={':value': {'S': ongoing_string}}
                )
                print(f"Updated item with partition key {item['id']} to 'ongoing': '{ongoing_string}'")
            else:
                # 'ongoing' attribute is already a string, continue
                print(f"'ongoing' attribute for item with partition key {item['id']} is already a string: '{item['ongoing']['S']}'")

# Call the function to update the values
update_ongoing_values()
